### PR TITLE
Update fclone.js

### PR DESCRIPTION
**Pull request by tracker1**
fix typedarray detection and regex.

### CODE REVIEW

File: src/fclone.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ function fclone(obj, refs) {
   }
 
   // typed array Int32Array etc.
-  if (/\d[a-z]*Array]$/i.test(Object.prototype.toString.call(obj))) {
+  if (obj.subarray  && typeof obj.subarray === 'function' && /^[A-Z][A-Za-z\d]+Array$/.test(Object.prototype.toString.call(obj))) {
```
Comment (soyuka) [inline src/fclone.js]: What was wrong with the old regex? 

Btw `typeof obj.subarray === 'function'` is sufficient, no need for `obj.subarray`

Comment (tracker1) [reply src/fclone.js]: Didn't mean to change the regexp, the subarray check was in order to verify the method before trying to call it, also, likely faster invalidation than the regex check.

File: src/fclone.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ function fclone(obj, refs) {
   }
 
   // typed array Int32Array etc.
-  if (obj.subarray  && typeof obj.subarray === 'function' && /^[A-Z][A-Za-z\d]+Array$/.test(Object.prototype.toString.call(obj))) {
+  if (typeof obj.subarray === 'function' && /[A-Z][A-Za-z\d]+Array/.test(Object.prototype.toString.call(obj))) {
```
Comment (soyuka) [inline src/fclone.js:23]: Removed start/stop flags on the regex?

Comment (tracker1) [reply src/fclone.js:23]:
```
> Object.prototype.toString.call(new Uint8Array())
'[object Uint8Array]'
```

It wasn't correct, as there's a literal `]` after the Array word...  Figure that if it has the subarray function and it has the word Array in the object string output, it's close enough...

Could make it `/^\[object [A-Z][A-Za-z\d]+Array\]$/` instead...

**Comment by soyuka** [comment]
Thanks @tracker1

**Pull request merged**